### DISCUSSION TITLE
Stored cross site scripting

### DISCUSSION THREAD

**Discussion by hellosmriti** [pimcore/pimcore#7839] [Q&A / Support]
I can store malicious payload into the affected area of the application and every user gets affected who browses through that malicious payload.

![Picture1](https://user-images.githubusercontent.com/54179597/103521513-e644cd00-4e9e-11eb-9caa-c2e6f0462c29.png)
![Picture2](https://user-images.githubusercontent.com/54179597/103521550-f52b7f80-4e9e-11eb-9cc3-fcb1ad712b07.png)
![Picture3](https://user-images.githubusercontent.com/54179597/103521558-f9f03380-4e9e-11eb-9063-9e5e791cd5a8.png)
![Picture4](https://user-images.githubusercontent.com/54179597/103521566-ff4d7e00-4e9e-11eb-95e4-020e40a77ba1.png)
![Picture5](https://user-images.githubusercontent.com/54179597/103521587-05435f00-4e9f-11eb-940c-8b45207255f6.png)

Any help is appreciated. Thanks in advance.

**Comment by NiklasBr** [comment]
I don't follow exactly what is happening because you are very light on details.

Is this correct?
1. You log into Pimcore with credentials which allow you to create and upload Assets.
2. You upload a file with the ".JPG" extension, but it is actually a ".SVG" file.
3. You send a HTTP request for the file in the browser.

Or are you doing this?
1. You log into Pimcore with credentials which allow you to create and upload Assets.
2. You upload a JPG file.
3. You upload an SVG file with the same file name via a repeated POST action.
4. You send a HTTP request for the SVG file in the browser.

Not clear to me. Can you explain in more details?

**Comment by hellosmriti** [comment]
Hi @NiklasBr ,

If you see the third screen-shot, you will see a script which we are trying to run by alerting out **document.domain**. And the script is getting executed when the file is opened in browser.

**Comment by NiklasBr** [comment]
Of course it is getting executed, it is a SVG file after all, and they, just like HTML files, can absolutely contain `<script>` elements with JavaScript code. What *specific* security issue have you discovered that should have prevented this?
